### ISSUE TITLE
HPatches评测问题

### ISSUE THREAD

**Issue by ljxg00dj0b** [leoluxxx/JamMa#39]
你好大佬，我用image-matching-toolbox进行HPatches评测，我把jamma加了进去，为什么会出现以下这种情况，无任何粗匹配，数据加载用的image-matching-toolbox自带的，数据预处理我是用的jamma的read_megadepth_color,调用了backbone和JamMa模块进行匹配，最后取data里面的"mkpts0_f"，"mkpts1_f"，"mconf"，但是在粗匹配阶段就出问题了。我的参数配置是粗thr0.3,细thr0.1,ransac阈值3，图像都padding为480*480的正方形，保存了掩码，匹配会输入掩码，jamma主程序无变动。数据预处理后保存的格式：data = { "imagec_0": rgb0,   "imagec_1": rgb1,  "mask0": mask0.unsqueeze(0),  "mask1": mask1.unsqueeze(0)}
我的理解是通过read_megadepth_color处理后img形状为（1，C，H，W），所以就没必要squeeze(0)，mask无批次大小所以添加unsqueeze(0)
 帅气的大佬能否提供一下HPatches评测的代码，我弄了好久都没搞定=.=！，谢谢啦🙇如果方便的话可以发我的邮箱（626807998@qq.com）
<img width="1554" height="373" alt="Image" src="https://github.com/user-attachments/assets/9f9b6ef9-ff6e-453e-8677-6a62dde2b676" />
以下是我的immatch.mudule的代码：

      class JamMa(Matching):
          def __init__(self, args):
              super().__init__()
              if type(args) == dict:
                  args = Namespace(**args)
      
              self.imsize = args.imsize
              self.match_threshold = args.match_threshold
              self.no_match_upscale = args.no_match_upscale
              self.eval_coarse = args.eval_coarse
      
              conf = dict(default_cfg)
              self.backbone = convnextv2_nano_()
              self.model = JamMa_(config=conf)
              ckpt_dict = torch.load(args.ckpt)
              
              # Separate backbone and matcher weights
              state_dict = ckpt_dict["state_dict"]
              backbone_state_dict = {k.replace('backbone.cnn.', ''): v for k, v in state_dict.items() if k.startswith('backbone.cnn.')}
              matcher_state_dict = {k.replace('matcher.', ''): v for k, v in state_dict.items() if k.startswith('matcher.')}
              # Load weights to respective models
              self.backbone.cnn.load_state_dict(backbone_state_dict, strict=True)
              self.model.load_state_dict(matcher_state_dict, strict=True)
              self.model = self.model.eval().to(self.device)
              self.backbone = self.backbone.eval().to(self.device)
      
              # Name the method
              self.ckpt_name = args.ckpt.split("/")[-1].split(".")[0]
              self.name = f"JamMa_{self.ckpt_name}"
              if self.no_match_upscale: self.name += "_noms"
              print(f"Initialize {self.name}")
      
          def load_im(self, im_path):
              return read_megadepth_color(im_path, self.imsize, 8, padding=True)
      
          def match_inputs_(self, rgb0, rgb1, mask0, mask1):
              rgb0 = rgb0.to(self.device)
              rgb1 = rgb1.to(self.device)
              mask0 = mask0.to(self.device)
              mask1 = mask1.to(self.device)
              data = {
                      "imagec_0": rgb0, 
                      "imagec_1": rgb1,
                      "mask0": mask0.unsqueeze(0),
                      "mask1": mask1.unsqueeze(0),
                      }
              self.backbone(data) 
              self.model(data)
      
              if self.eval_coarse:
                  kpts1 = data["mkpts0_c"].cpu().numpy()
                  kpts2 = data["mkpts1_c"].cpu().numpy()
              else:
                  kpts1 = data["mkpts0_f"].cpu().numpy()
                  kpts2 = data["mkpts1_f"].cpu().numpy()
              scores = data["mconf"].cpu().numpy()  
              matches = np.concatenate([kpts1, kpts2], axis=1)
              return matches, kpts1, kpts2, scores
      
          def match_pairs(self, im1_path, im2_path):
              rgb0, scale0, mask0, _ = self.load_im(im1_path)
              rgb1, scale1, mask1, _ = self.load_im(im2_path)
      
              [ts_mask_0, ts_mask_1] = F.interpolate(torch.stack([mask0, mask1], dim=0)[None].float(),
                                                             scale_factor=0.125, mode='nearest',  recompute_scale_factor=False)[0].bool()
                  
              matches, kpts1, kpts2, scores = self.match_inputs_(rgb0, rgb1, ts_mask_0, ts_mask_1)
              sc0 = scale0.cpu().numpy()
              sc1 = scale1.cpu().numpy()
              upscale = np.array([np.concatenate([sc0, sc1])])
      
              if self.no_match_upscale:
                  return matches, kpts1, kpts2, scores, upscale.squeeze(0)
      
              # Upscale matches &  kpts
              matches = upscale * matches
              kpts1 = sc0 * kpts1
              kpts2 = sc1 * kpts2
              return matches, kpts1, kpts2, scores
